mvc/src/main/java/nextstep/mvc/controller/tobe/ControllerScanner.java

```diff
@@ -0,0 +1,35 @@
+package nextstep.mvc.controller.tobe;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import nextstep.web.annotation.Controller;
+import org.reflections.Reflections;
+
+public class ControllerScanner {
+
+    private final Reflections reflections;
+
+    public ControllerScanner(final Object... basePackage) {
+        this.reflections = new Reflections(basePackage);
+    }
+
+    public Map<Class<?>, Object> getControllers() {
+        final Set<Class<?>> controllerClasses = reflections.getTypesAnnotatedWith(Controller.class);
+        try {
+            Map<Class<?>, Object> controllers = new HashMap<>();
+            instantiateControllers(controllerClasses, controllers);
+            return controllers;
+        } catch (final InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException();
+        }
+    }
+
+    private void instantiateControllers(final Set<Class<?>> controllerClasses, final Map<Class<?>, Object> controllers)
+            throws InstantiationException, IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        for (Class<?> controllerClass : controllerClasses) {
+            controllers.put(controllerClass, controllerClass.getDeclaredConstructor().newInstance());
+        }
+    }
```

Comment:
```suggestion
    public Map<Class<?>, Object> getControllers() {
        try {
            return instantiateControllers(reflections.getTypesAnnotatedWith(Controller.class));
        } catch (final InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
            throw new RuntimeException();
        }
    }

    private  Map<Class<?>, Object> instantiateControllers(final Set<Class<?>> controllerClasses)
            throws InstantiationException, IllegalAccessException, InvocationTargetException, NoSuchMethodException {
        Map<Class<?>, Object> controllers = new HashMap<>();
        for (Class<?> controllerClass : controllerClasses) {
            controllers.put(controllerClass, controllerClass.getDeclaredConstructor().newInstance());
        }
        return controllers;
    }
```

c: 이런식으로 처리할 수도 있겠네요! 메서드의 파라미터가 줄어서 더 깔끔한 것 같아요

Reply:
오호 요 방법이 더 괜찮네요 3단계때 반영하겠습니다